base/file.jl

```diff
@@ -517,9 +517,9 @@ created. The path is likely to be unique, but this cannot be guaranteed.
 
 !!! warning
 
-    This can lead to race conditions if another process obtains the same
-    file name and creates the file before you are able to.
-    Using [`mktemp()`](@ref) is recommended instead.
+    This can lead to security holes if another process obtains the same
+    file name and creates the file before you are able to. Open the file with
+    `JL_O_EXCL` if this is a concern. Using [`mktemp()`](@ref) is also recommended instead.
```

Comment:
Also open an issue that the implementation of `mktemp` on windows is wrong?

Reply:
Unrelated: It's  a mystery why we have our own `mktempdir` implementation on windows where the linux mktempdir function works cross-platform.  Is this just an oversight?

Reply:
Yeah we should  just use  https://github.com/libuv/libuv/pull/2074 when it gets through review to fix the Windows implementation.